apps/condo/domains/resident/schema/DiscoverServiceConsumersService.test.js

```diff
@@ -3,275 +3,519 @@
+            const residentClient1 = await makeClientWithResidentAccessAndProperty()
+            const residentClient2 = await makeClientWithResidentAccessAndProperty()
+
+            const [billingIntegration] = await createTestBillingIntegration(admin)
+            const [billingContext] = await createTestBillingIntegrationOrganizationContext(admin, residentClient1.organization, billingIntegration, { status: CONTEXT_FINISHED_STATUS })
+            const [acquiringIntegration] = await createTestAcquiringIntegration(admin)
+            await createTestAcquiringIntegrationContext(admin, residentClient1.organization, acquiringIntegration, { status: CONTEXT_FINISHED_STATUS })
+
+            const unitType1 = FLAT_UNIT_TYPE
+            const unitName1 = faker.lorem.word()
+
+            // 1/3 register residents
+            await registerResidentByTestClient(
+                residentClient1,
+                {
+                    address: residentClient1.property.address,
+                    addressMeta: residentClient1.property.addressMeta,
+                    unitType: unitType1,
+                    unitName: unitName1,
+                })
+
+            await registerResidentByTestClient(
+                residentClient2,
+                {
+                    address: residentClient1.property.address,
+                    addressMeta: residentClient1.property.addressMeta,
+                    unitType: unitType1,
+                    unitName: unitName1,
+                })
+
+            // 2/3 upload receipts
+            const payload = {
+                context: { id: billingContext.id },
+                receipts: [
+                    createRegisterBillingReceiptsPayload({
+                        address: residentClient1.property.address,
+                        unitType: unitType1,
+                        unitName: unitName1,
+                    }),
+                    createRegisterBillingReceiptsPayload(),
+                ],
+            }
+            const [registeredReceipts] = await registerBillingReceiptsByTestClient(admin, payload)
+
+            // 3/3 check that residents can see the receipts list
+            await waitFor(async () => {
+                const receipts1 = await ResidentBillingReceipt.getAll(residentClient1, {})
+                const receipts2 = await ResidentBillingReceipt.getAll(residentClient2, {})
+
+                expect(receipts1).toHaveLength(1)
+                expect(registeredReceipts).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: receipts1[0].id }),
+                ]))
+
+                expect(receipts2).toHaveLength(1)
+                expect(registeredReceipts).toEqual(expect.arrayContaining([
+                    expect.objectContaining({ id: receipts2[0].id }),
+                ]))
+            })
         })
     })
```

Comment:
1. Org -> resident
2. Provider -> resident
3. resident -> provider
4. resident -> org

Reply:
The new test was added